fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabase.java

```diff
@@ -1120,6 +1134,55 @@ protected RuntimeException mapAsyncToSyncException(@Nonnull Throwable ex) {
         return future.get();
     }
 
+    /**
+     * Log warning and close tracked contexts that have been open for too long.
+     * @param minAgeSeconds number of seconds above which to warn
+     * @return number of such contexts found
+     */
+    public int warnAndCloseOldTrackedOpenContexts(long minAgeSeconds) {
+        long nanoTime = System.nanoTime() - TimeUnit.SECONDS.toNanos(minAgeSeconds);
+        if (trackedOpenContexts.isEmpty()) {
+            return 0;
+        }
+        try {
+            if (trackedOpenContexts.firstKey() > nanoTime) {
```

Comment:
I think if this used `pollFirstEntry`, then this wouldn't need to catch an exception (but would have to check for null). I suppose that could then be combined with the check to see if the data structure is empty.

Reply:
You mean `firstEntry`? We don't want to remove it from the map. I assume that the most common case is that the map is not empty but all entries are younger than the threshold. `firstEntry` conses. I suppose it'll probably be on the stack, so maybe this isn't a real issue.